src/NServiceBus.Core/Sagas/Sagas.cs

```diff
@@ -98,6 +98,13 @@ static bool IsCompatible(Type t, Type source)
 
         static bool IsTypeATimeoutHandledByAnySaga(Type type, IEnumerable<Type> sagas)
         {
+            // MakeGenericType() throws an exception if passed a ref struct type
+            // Messages cannot be ref struct types
+            if (type.CustomAttributes.Any(a => a.AttributeType.FullName == "System.Runtime.CompilerServices.IsByRefLikeAttribute"))
```

Comment:
Should we use `string.Equals()` for string comparison?

Reply:
would that be for performance reasons? 

https://stackoverflow.com/questions/1659097/why-would-you-use-string-equals-over suggests that there is no difference in behaviour?

Reply:
[They lead to the same code path](https://github.com/dotnet/runtime/blob/5535e31a712343a63f5d7d796cd874e563e5ac14/src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs#L688-L701). What you save is the operator method invocation cost